pkg/controller/machine_safety_test.go

```diff
@@ -71,6 +71,121 @@ var _ = Describe("machine", func() {
 		Entry("one machineset", nil, map[string]string{"machineset-1": ""}, nil, []string{"machineset-1"}, nil),
 	)
 
+	DescribeTable("##unfreezeMachineSetsAndDeployments",
```

Comment:
Cant comment at the moment on the test. Would be really great, if we can have a session soon on how are we introducing the tests overall for the repo with details.

Reply:
@amshuman-kr plans to give a session sometime this week. Do you  want to keep this PR open until then?

Reply:
Scheduled for Thursday 11am.